### PR TITLE
Rename ctx.Form() to ctx.FormString() and move code into own file

### PR DESCRIPTION
**Pull request by 6543**
Followup from #16562 prepare for #16567

* [x] Rename ctx.Form() to ctx.FormString()
* [x] Reimplement FormX func to need less code and cpu cycles
* [x] Move code into own file

### CODE REVIEW

File: modules/context/form.go [hunk 1]
```diff
@@ -1,241 +1,64 @@
-// Copyright 2021 The Gitea Authors. All rights reserved.
+// Copyright 2014 The Gogs Authors. All rights reserved.
+// Copyright 2020 The Gitea Authors. All rights reserved.
```
Comment (6543) [inline modules/context/form.go]:
```suggestion
// Copyright 2021 The Gitea Authors. All rights reserved.
```

File: routers/web/user/auth_openid.go [hunk 2]
```diff
@@ -129,7 +129,7 @@ func SignInOpenIDPost(ctx *context.Context) {
 	url += "&openid.ns.sreg=http%3A%2F%2Fopenid.net%2Fextensions%2Fsreg%2F1.1"
 	url += "&openid.sreg.optional=nickname%2Cemail"
 
-	log.Trace("Form-passed openid-remember: %t", form.Remember)
+	log.Trace("FormString-passed openid-remember: %t", form.Remember)
```
Comment (6543) [inline routers/web/user/auth_openid.go]:
```suggestion
	log.Trace("Form-passed openid-remember: %t", form.Remember)
```

**Review by lunny — APPROVED** [review]

**Comment by zeripath** [comment]
I don't understand the reasoning behind renaming form to formstring.

It appears to make it inconsistent with the way we use Query and with ini.

Is the intention that form will return the form in the future?

**Comment by 6543** [comment]
@zeripath the problem is that `ctx.Req.Form` != `ctx.Form` at the moment ... a name is colliding :/

to the form thing ... it's golang specific - i personaly also prevere to use the term query ... but net/http do differ between GET and POST requests - Query only apply to get where Form do the same and apply to all

File: modules/context/form.go [hunk 3]
```diff
@@ -5,237 +5,60 @@
...
+func (ctx *Context) FormBool(key string) bool {
+	v, _ := strconv.ParseBool(ctx.Req.FormValue(key))
 	return v
 }
 
-// MustOptionalBool returns request form as OptionalBool with default
-func (f *Forms) MustOptionalBool(key string, defaults ...util.OptionalBool) util.OptionalBool {
-	value := (*http.Request)(f).FormValue(key)
+// FormOptionalBool returns an OptionalBoolTrue or OptionalBoolFalse if value for key exist
+// else it return OptionalBoolNone
```
Comment (zeripath) [inline modules/context/form.go]:
```suggestion
// FormOptionalBool returns an OptionalBoolTrue or OptionalBoolFalse if the value 
// for the provided key exists in the form else it returns OptionalBoolNone
```

**Comment by 6543** [comment]
@zeripath can you `go fmt` for me please?

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
@zeripath I'll merge this for now, if we like to go a interface way as you proposed ... I'm happy to review it. For now I think this is god to go and we can move things forward ... (It's already cleaner this way & rm useless code)

**Pull request merged**